### PR TITLE
Assert rep_time is properly handled in backendconfiguration tests

### PR DESCRIPTION
**Pull request by scottwn**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.

### Summary
Previously the Integer field in the pulse backend schema was rounding rep_times to 0. This has been fixed by other recent changes, but to avoid regressing it in the future this PR adds on to tests for the backend configuration to assert this continues working moving forward.

### CODE REVIEW

**Comment by taalexander** [comment]
The jsonschema is expecting rep_time to be an integer https://github.com/Qiskit/qiskit-terra/blob/8c20688ebd400e18ffec9dfb2815330ff1502557/qiskit/schemas/qobj_schema.json#L1024. We should change this to be a `number` to resolve this issue properly. Does it also seem like your PR is only modifying tests currently?

**Comment by scottwn** [comment]
This PR had more changes to `backendconfiguration.py` and the fake backends, but I removed them because they conflicted with #4016. (Those changes are still at https://github.com/Qiskit/qiskit-terra/compare/master..scottwn:issue-3920-conflicts)

The problem described in #3920 is caused by https://github.com/Qiskit/qiskit-terra/blob/52d394a132b98efc71d801bbeb0358cc83b5b890/qiskit/providers/models/backendconfiguration.py#L118

Changing `Integer` to `Float` fixes the problem. I can't find where `qobj_schema.json` is used in the `configuration` flow; `validate_json_against_schema` doesn't look like it's called outside of the test modules and after changing the schema to `number` the test still fails. Can you point me to where the jsonschema is checked?

**Comment by taalexander** [comment]
The schema is used to validate on the API side. I agree with your fix on the Qiskit side. I just think that if we are converting the integer to a float, all references to `rep_time` and `rep_times` in the schemas must also be updated to floats to maintain compatibility. A `ctrl+f` of Qiskit should suffice to find these.

**Comment by scottwn** [comment]
I think part of the issue here is that `rep_time` is represented as seconds in some places and microseconds in other places. The issue of rounding `rep_time` to 0 (described in #3920, fixed in #4016, with a test case in this PR) happens after `backendconfiguration.py` converts from microseconds to seconds. https://github.com/Qiskit/qiskit-terra/blob/4a74e27c03bf3ee3249b5627072581f6a93a7888/qiskit/providers/models/backendconfiguration.py#L453
After the conversion, the configuration was still expecting an integer and rounding to 0.

The jsonschema is used to validate in modules like `pulse_qobj.py` where `rep_time` is only represented in microseconds (as far as I can tell). `_parse_pulse_args` converts to microseconds before the jsonschema is validated. https://github.com/Qiskit/qiskit-terra/blob/4a74e27c03bf3ee3249b5627072581f6a93a7888/qiskit/compiler/assemble.py#L280
So I think we want the jsonschema to remain `"type": "integer"` unless we're allowing `rep_time` less than 1 microsecond.

**Comment by taalexander** [comment]
Yes, I agree with you. 

> I think part of the issue here is that `rep_time` is represented as seconds in some places and microseconds in other places. The issue of rounding `rep_time` to 0 (described in #3920, fixed in #4016, with a test case in this PR) happens after `backendconfiguration.py` converts from microseconds to seconds.
> 
> https://github.com/Qiskit/qiskit-terra/blob/4a74e27c03bf3ee3249b5627072581f6a93a7888/qiskit/providers/models/backendconfiguration.py#L453
> 
> 
> After the conversion, the configuration was still expecting an integer and rounding to 0.
> The jsonschema is used to validate in modules like `pulse_qobj.py` where `rep_time` is only represented in microseconds (as far as I can tell). `_parse_pulse_args` converts to microseconds before the jsonschema is validated.
> 
> https://github.com/Qiskit/qiskit-terra/blob/4a74e27c03bf3ee3249b5627072581f6a93a7888/qiskit/compiler/assemble.py#L280
> 
> 
> So I think we want the jsonschema to remain `"type": "integer"` unless we're allowing `rep_time` less than 1 microsecond.

Yes, you're right 😄

File: test/python/providers/test_backendconfiguration.py [hunk 1]
```diff
@@ -110,4 +112,4 @@ def test_get_channel_prefix_index(self):
 
     def _test_lists_equal(self, actual, expected):
         """Test if 2 lists are equal. It returns ``True`` is lists are equal."""
-        return collections.Counter(actual) == collections.Counter(expected)
+        return collections.Counter(actual) == collections.Counter(expected)
```
Comment (taalexander) [inline test/python/providers/test_backendconfiguration.py:115]: It seems you removed the newline at the end of file?

Comment (scottwn) [reply test/python/providers/test_backendconfiguration.py:115]: My mistake, thanks for catching that....

**Pull request merged**